### PR TITLE
feat(compiler-vapor): `v-else` / `v-else-if`

### PR DESCRIPTION
**Pull request by LittleSound**
related: #9
compiler for `v-if`: #96 
`createIf` (`v-if` of runtime-vapor): #95

### CODE REVIEW

File: packages/compiler-vapor/src/transforms/vIf.ts [hunk 1]
```diff
@@ -52,37 +54,109 @@ export function processIf(
...
+    // check the adjacent v-if
+    const siblings = context.parent!.node.children
+    const comments = []
+    let sibling: TemplateChildNode | undefined
+    let i = siblings.indexOf(node)
+    while (i-- >= -1) {
+      sibling = siblings[i]
+
+      if (sibling && sibling.type === NodeTypes.COMMENT) {
+        context.removeNode(sibling)
```
Comment (sxzz) [inline packages/compiler-vapor/src/transforms/vIf.ts:67]: AST should not be mutated. Unlike old compiler, we only need to generate IR node, and keep AST not changed.

File: packages/compiler-vapor/src/transforms/vIf.ts [hunk 2]
```diff
@@ -52,37 +54,109 @@ export function processIf(
...
+        __DEV__ && comments.unshift(sibling)
+        continue
+      }
+
+      if (
+        sibling &&
+        sibling.type === NodeTypes.TEXT &&
+        !sibling.content.trim().length
+      ) {
+        context.removeNode(sibling)
```
Comment (sxzz) [inline packages/compiler-vapor/src/transforms/vIf.ts:77]: Just skip it and don't remove it

**Review by sxzz — CHANGES_REQUESTED** [review]
It is worth noting that in Vapor, we no longer modify the AST directly, but generate a new IR structure. Therefore, all logic related to `removeNode` should be removed.

File: packages/compiler-vapor/src/transforms/vIf.ts (hunk 2 above)
Comment (LittleSound) [reply packages/compiler-vapor/src/transforms/vIf.ts:77]: Not deleting nodes can lead to some other problems. You can reproduce the problem by removing these codes and then running unit tests.

**Review by sxzz — APPROVED** [review]

**Comment by LittleSound** [comment]
Thank you for completing this PR for me while I was resting, I really appreciate it.

**Pull request merged**